slice out of bounds with peek > buffer size after partial read

This is a bit tricky and it took quite a while to find a simple repo case. The
following code panics with slice out of bounds:

```go
package main

import (
	"bytes"
	"fmt"

	"github.com/philhofer/fwd"
)

func main() {
	r := fwd.NewReader(bytes.NewReader(make([]byte, 40)))
	b := make([]byte, 21)
	_, err := r.Read(b)
	if err != nil {
		fmt.Println(err)
		return
	}
	r.Peek(2049)
}
```

```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/philhofer/fwd.(*Reader).more(0xc82003bef8)
	go/src/github.com/philhofer/fwd/reader.go:102 +0x27f
github.com/philhofer/fwd.(*Reader).Peek(0xc82003bef8, 0x801, 0x0, 0x0, 0x0, 0x0, 0x0)
	go/src/github.com/philhofer/fwd/reader.go:159 +0x1c3
main.main()
	t.go:18 +0x3b3
exit status 2
```

I'm still not clear what are the exact conditions (e.g. if you only read 20
bytes it won't panic, peek only 2048 and it won't panic). I think there is
something wrong with the reallocation logic in Peek(). The following patch
_seems_ to be correct and fix this, but then again maybe it just shifts around
the trigger condition a bit.

```diff
diff --git a/reader.go b/reader.go
index ba07f12..c854957 100644
--- a/reader.go
+++ b/reader.go
@@ -150,6 +150,7 @@ func (r *Reader) Peek(n int) ([]byte, error) {
                old := r.data[r.n:]
                r.data = make([]byte, n+r.buffered())
                r.data = r.data[:copy(r.data, old)]
+               r.n = 0
        }
 
        // keep filling until
```

--- a/reader.go
+++ b/reader.go
@@ -148,6 +148,7 @@ func (r *Reader) Peek(n int) ([]byte, error) {
 		old := r.data[r.n:]
 		r.data = make([]byte, n+r.buffered())
 		r.data = r.data[:copy(r.data, old)]
+		r.n = 0
 	}
 
 	// keep filling until

